mpqp/core/instruction/gates/native_gates.py

```diff
@@ -741,7 +847,19 @@ def to_other_language(
                     "export, this feature is coming very soon!"
                 )
 
-            return braket_U(self.theta, self.phi, self.gamma)
+            return self.braket_gate(self.theta, self.phi, self.gamma)
+        if language == Language.QASM2:
+            from mpqp.qasm.mpqp_to_qasm import float_to_qasm_str
+
+            instruction_str = self.qasm2_gate
+            instruction_str += (
+                "("
+                + ",".join(float_to_qasm_str(float(param)) for param in self.parameters)
+                + ")"
+            )
+            qubits = ",".join([f"q[{j}]" for j in self.targets])
+
+            return instruction_str + " " + qubits + ";"
         else:
             raise NotImplementedError(f"Error: {language} is not supported")
```

Comment:
Here a comment on the translation to OpenQASM3 for a circuit containing the gate `U`.

We pass through openqasm2 that will transform the gate `U`to an openQasm2 `u` (with potentially handling of the global phase).

However, when we will translate to openqasm3, we will consider that this is a gate `u` (without nowing it was coming from a `U` gate in mpqp that is the same as the openqasm3 `U`), and so it will translate it to the openqasm3 gate `u3`, while it should have been the native keyword `U` in openQASM3, and this will generate for sure a global phase problem.

This today may affect `AWSDevice`s since we use that pipeline to translate. We should in the future do the translation directly to OpenQASM3 if we can.

One can verify this statement with the following code: 
```Python
>>> from mpqp.all import *
>>> c = QCircuit([U(1,2,3,0)])
>>> print(c.to_other_language(Language.QASM3))
OPENQASM 3.0;
include "stdgates.inc";

qubit[1] q;
u3(1,2,3) q[0];
```

If one checks the definition of `u3` (and its matrix) and compare with the mpqp `U`, he will witness the problem.